src/subscription_manager/cli_command/register.py

```diff
@@ -316,56 +316,57 @@ def _prompt_for_environment(self):
+                environments = options.environments
+            else:
                 # If there aren't any environments, don't prompt for one
-                if not env_list:
-                    return environment_name
+                if not master_env_list:
+                    return None
 
                 # If the envronment list is len 1, pick that environment
-                if len(env_list) == 1:
-                    log.debug("Using the only available environment: \"{name}\"".format(name=env_list[0]["name"]))
-                    return env_list[0]['id']
+                if len(master_env_list) == 1:
+                    log.debug("Using the only available environment: \"{name}\"".format(name=master_env_list[0]["name"]))
+                    return master_env_list[0]['id']
 
-                env_name_list = [env['name'] for env in env_list]
+                env_name_list = [env['name'] for env in master_env_list]
                 print(_('Hint: Organization "{key}" contains following environments: {list}').format(
                       key=owner_key, list=", ".join(env_name_list)))
-
-                environment_name = self._prompt_for_environment()
-
-                # Should only ever be len 0 or 1
-                env_matches = [env['id'] for env in env_list if env['name'] == environment_name]
-                if env_matches:
-                    return env_matches[0]
-                system_exit(os.EX_DATAERR, _("No such environment: {name}").format(name=environment_name))
-
-            # Server doesn't support environments
-            return environment_name
-
-        if not supports_environments:
-            system_exit(os.EX_UNAVAILABLE, _("Error: Server does not support environments."))
-
-        env = cp.getEnvironment(owner_key=owner_key, name=environment_name)
-        if not env:
-            system_exit(os.EX_DATAERR, _("No such environment: {name}").format(name=environment_name))
-        return env['id']
+                environments = self._prompt_for_environment()
+
+            # We can have multiple
+            names = environments.split(",")
+            first = True
+            for name in names:
+                env_id = [env['id'] for env in master_env_list if env['name'] == name]
+                if not env_id:
+                    system_exit(os.EX_DATAERR, _("No such environment: {name}").format(name=name))
+                else:
+                    if first:
+                        first = False
+                    else:
+                        environment_ids = environment_ids + ","
+                        if name in environment_set:
+                            system_exit(os.EX_DATAERR, _("Error: The environment '{name}' cannot be listed twice. ").format(name=name))
+                    environment_set.add(name)
+                    environment_ids += env_id[0]
+
+        return environment_ids
 
     @staticmethod
     def _no_owner_cb(username):
```

Comment:
In addition to the changes in register and environment commands, we also need to change `cli_command/identity.py` to print the list of environments the consumer is a part of (since currently it does list the environment of the consumer, if a part of one).